src/arrayMethodJoin.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let between = separator;
+
+    if (separator === undefined) {
+      between = ',';
+    }
```

Comment:
Actually, you don't need this part of code, you can just set the default value for separator in the function params list.